### PR TITLE
healthcare API: add AttributeDefinition CRUD samples+tests

### PR DESCRIPTION
**Pull request by noerog**
## Description

Part of b/224797364.

## Checklist
- [X] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [X] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [X] **Tests** pass:   `nox -s py-3.6` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [X] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [X] Please **merge** this PR for me once it is approved.

### CODE REVIEW

File: healthcare/api-client/v1/consent/attribute_definitions.py [hunk 1]
```diff
@@ -0,0 +1,494 @@
...
+    response = request.execute()
+    print("Created RESOURCE attribute definition: {}".format(response))
+
+    return response
+
+
+# [END healthcare_create_resource_attribute_definition]
+
+
+# [START healthcare_create_resource_attribute_definition]
```
Comment (busunkim96) [inline healthcare/api-client/v1/consent/attribute_definitions.py:81]: Should this be `create_request`? It has the same region tag as the function above.

**Review by busunkim96 — APPROVED** [review]

File: healthcare/api-client/v1/consent/attribute_definitions.py (hunk 1 above)
Comment (noerog) [reply healthcare/api-client/v1/consent/attribute_definitions.py:81]: Yes, thanks for catching.

**Comment by noerog** [comment]
I pinged @[dandhlee](https://github.com/dandhlee) to get the IAM permissions configured on the testing projects which should fix the failing tests.

**Comment by dandhlee** [comment]
Apologies for the delay, I'll take a look at this first thing in the morning.

File: healthcare/api-client/v1/consent/README.rst [hunk 2]
```diff
@@ -12,12 +11,14 @@ This directory contains samples for Cloud Healthcare API. `Cloud Healthcare API`
...
-.. _Cloud Healthcare API: https://cloud.google.com/healthcare/docs
+.. _Cloud Healthcare API: https://cloud.google.com/healthcare-api/docs
 
 To run the sample, you need to enable the API at: https://console.cloud.google.com/apis/library/healthcare.googleapis.com
 
-To run the sample, you need to have `Healthcare Consent Store Administrator` role.
 
+To run the sample, you need to have the following roles:
+* `Healthcare Consent Store Administrator`
+* `Healthcare Attribute Definition Editor`
```
Comment (dandhlee) [inline healthcare/api-client/v1/consent/README.rst:21]: These show up all bundled in the same line, I think you might need an extra space?

File: healthcare/api-client/v1/consent/README.rst [hunk 3]
```diff
@@ -41,81 +42,77 @@ Install Dependencies
 #. Clone python-docs-samples and change directory to the sample directory you want to use.
 
     .. code-block:: bash
-
         $ git clone https://github.com/GoogleCloudPlatform/python-docs-samples.git
```
Comment (dandhlee) [inline healthcare/api-client/v1/consent/README.rst:45]: The code doesn't seem to show anymore, seems that the newlines are needed.

File: healthcare/api-client/v1/consent/README.rst [hunk 4]
```diff
@@ -41,81 +42,77 @@ Install Dependencies
...
-   :target: https://console.cloud.google.com/cloudshell/open?git_repo=https://github.com/GoogleCloudPlatform/python-docs-samples&page=editor&open_in_editor=healthcare/api-client/v1/consents/consent_stores.py,healthcare/api-client/v1/consents/README.rst
+   :target: https://console.cloud.google.com/cloudshell/open?git_repo=https://github.com/GoogleCloudPlatform/python-docs-samples&page=editor&open_in_editor=healthcare/api-client/v1/consent/consent_stores.py,healthcare/api-client/v1/consent/README.rst
 
 
 
 
 To run this sample:
 
 .. code-block:: bash
-
```
Comment (dandhlee) [inline healthcare/api-client/v1/consent/README.rst:83]: Ditto, codeblock does not show up

File: healthcare/api-client/v1/consent/README.rst [hunk 5]
```diff
@@ -129,26 +126,101 @@ To run this sample:
...
 
+.. image:: https://gstatic.com/cloudssh/images/open-btn.png
+   :target: https://console.cloud.google.com/cloudshell/open?git_repo=https://github.com/GoogleCloudPlatform/python-docs-samples&page=editor&open_in_editor=healthcare/api-client/v1/consent/attribute_definitions.py,healthcare/api-client/v1/consent/README.rst
 
 
 
 
+To run this sample:
+
+.. code-block:: bash
```
Comment (dandhlee) [inline healthcare/api-client/v1/consent/README.rst:161]: Ditto, code block does not show up.

File: healthcare/api-client/v1/consent/README.rst.in [hunk 6]
```diff
@@ -10,7 +10,9 @@ product:
     data with cloud-based applications.
 
 required_api_url: https://console.cloud.google.com/apis/library/healthcare.googleapis.com
-required_role: Healthcare Consent Store Administrator
```
Comment (dandhlee) [inline healthcare/api-client/v1/consent/README.rst.in:13]: Hmm... I wonder if the readme generator broke? Doesn't seem like you did anything particularly different to break so much of the readme...

File: healthcare/api-client/v1/consent/attribute_definitions_test.py [hunk 7]
```diff
@@ -0,0 +1,210 @@
+# Copyright 2022 Google LLC
```
Comment (dandhlee) [inline healthcare/api-client/v1/consent/attribute_definitions_test.py:1]: Any test cases for `patch_attribute_definition`?

**Review by dandhlee — COMMENTED** [review]
Python style wise looks good to me, few comments on the README for the test.

File: healthcare/api-client/v1/consent/attribute_definitions_test.py (hunk 7 above)
Comment (noerog) [reply healthcare/api-client/v1/consent/attribute_definitions_test.py:1]: Done.

File: healthcare/api-client/v1/consent/README.rst (hunk 5 above)
Comment (noerog) [reply healthcare/api-client/v1/consent/README.rst:176]: Fixed.

File: healthcare/api-client/v1/consent/README.rst (hunk 4 above)
Comment (noerog) [reply healthcare/api-client/v1/consent/README.rst]: Fixed.

File: healthcare/api-client/v1/consent/README.rst (hunk 3 above)
Comment (noerog) [reply healthcare/api-client/v1/consent/README.rst]: Fixed.

**Review by dandhlee — APPROVED** [review]
LGTM, one minor comment below.

**Pull request merged**